source/src/BNFC/Backend/FSharp.hs

```diff
@@ -0,0 +1,246 @@
+fsyaccFileM   = mkMod  withLang "Par"
+templateFile  = mkFile withLang "Skel" "fs"
+templateFileM = mkMod  withLang "Skel"
+printerFile   = mkFile withLang "Print" "fs"
+printerFileM  = mkMod  withLang "Print"
+showFile      = mkFile withLang "Show" "fs"
+showFileM     = mkMod  withLang "Show"
+tFileM        = mkMod  withLang "Test"
+tFile         = mkFile withLang "Test" "fs"
+utilFileM     = mkMod  noLang   "BnfcUtil"
+utilFile      = mkFile noLang   "BnfcUtil" "fs"
+fsprojFile    = mkFile withLang  "" "fsproj"
+buildTarget   = mkFile withLang  "" ""
+
+makeFSharp :: SharedOptions -> CF -> MkFiles ()
+makeFSharp opts cf = do
+  let absMod = absFileM opts
+      lexMod = fslexFileM opts
+      parMod = fsyaccFileM opts
+      prMod  = printerFileM opts
+      showMod = showFileM opts
+      tFileMod = tFileM opts
+  do
+    mkfile (absFile opts)       comment $ cf2Abstract absMod cf
+    mkfile (fslexFile opts)  comment $ cf2fslex lexMod parMod cf
+    mkfile (fsyaccFile opts) C.comment $
+                 cf2fsyacc parMod absMod lexMod  cf
+    mkfile (templateFile opts)  comment $ cf2Template (templateFileM opts) absMod cf
+    mkfile (printerFile opts)   comment $ cf2Printer prMod absMod cf
+    mkfile (showFile opts)      comment $ cf2show showMod absMod cf
+    mkfile (tFile opts)         comment $ fsharpTestfile absMod lexMod parMod prMod showMod tFileMod cf
+    mkfile (utilFile opts)      comment $ utilM (utilFileM opts)
+    mkfile (fsprojFile opts)    XML.comment $ fsprojM opts
+    mkMakefile opts $ makefile opts
+    -- case xml opts of
+    --   2 -> makeXML opts True cf
+    --   1 -> makeXML opts False cf
+    --   _ -> return ()
+
+-- | Generate the makefile.
+makefile
+  :: SharedOptions
+  -> String    -- ^ Filename of the makefile.
+  -> Doc       -- ^ Content of the makefile.
+makefile opts makeFile = vcat
+  [ "# Makefile for building the parser and test program."
+  , phonyRule
+  , defaultRule
+  , vcat [ "# Rules for building the parser." , "" ]
+  -- If option -o was given, we have no access to the grammar file
+  -- from the Makefile.  Thus, we have to drop the rule for
+  -- reinvokation of bnfc.
+  , when (isDefault outDir opts) $ bnfcRule
+  , testParserRule
+  , vcat [ "# Rules for cleaning generated files." , "" ]
+  , cleanRule
+  , distCleanRule
+  , "# EOF"
+  ]
+  where
+  -- | List non-file targets here.
+  phonyRule :: Doc
+  phonyRule = vcat
+    [ "# List of goals not corresponding to file names."
+    , ""
+    , Makefile.mkRule ".PHONY" [ "all", "clean", "distclean" ] []
+    ]
+  -- | Default: build test parser(s).
+  defaultRule :: Doc
+  defaultRule = vcat
+     [ "# Default goal."
+     , ""
+     , Makefile.mkRule "all" tgts []
+     ]
+     where
+     tgts = [ buildTarget opts ]
+
+  -- | Rule to build F# test parser.
+  testParserRule :: Doc
+  testParserRule = Makefile.mkRule tgt deps [ "dotnet build" ]
+    where
+    tgt :: String
+    tgt = buildTarget opts
```

Comment:
I used a name that is not a file name. Is this a problem?

Reply:
Well, if the target is a file, then `make` can use its time-stamp to decide on rebuilding.
But I guess you have your reasons to not put a file there (maybe it is too complicated or fragile).
I am OK with that.

Reply:
For a moment I believed that there won't be a common file name for Linux and Windows but this turned out to be false. I will fix it.